waiter/src/waiter/handler.clj

```diff
@@ -1282,3 +1282,96 @@
         request))
     (catch Throwable th
       (utils/exception->response th request))))
+
+
+(defn get-service-id-from-instance-id [instance-id]
```

Comment:
we should pass in the service id explicitly and not couple calculations from the instance id.